core/src/tasks/profile/sweep.ts

```diff
@@ -39,13 +42,12 @@ export class ProfileSweep extends CLSTask {
       }
     );
 
-    // use "destroy" to clean up related models
     if (profiles.length > 0) {
-      await Profile.destroy({
-        where: {
-          id: profiles.map((p) => p.id),
-        },
-      });
+      const profileIds = profiles.map((p) => p.id);
+      await Profile.destroy({ where: { id: profileIds } });
+      await Import.destroy({ where: { profileId: profileIds } });
+      await Export.destroy({ where: { profileId: profileIds } });
+      await ProfileProperty.destroy({ where: { profileId: profileIds } });
     }
```

Comment:
I think I'd be worried about forgetting to add another relationship here in the future.  Since this is in a task, for now, I think `for (const profile of profiles) await profile.destroy()` would be OK... 

In the future, perhaps we make an AfterBatchDestroy which then uses the other hooks, keeping all the destroy logic within the model.